### DISCUSSION TITLE
Unloading resources

### DISCUSSION THREAD

**Discussion by jeffw387** [bevyengine/bevy#2671] [Q&A] [answered]
I'm considering how I would load and unload large images as required (based on player location). I think this might be called texture streaming or something similar? Not sure though.

Is there a way to deliberately unload an asset that isn't currently needed? Is this already done in the background when assets aren't being used (due to memory pressure or similar)?

I can see that the AssetServer has the free_unused_assets() function, but I'm not sure if that's what I'm looking for or not.

**Comment by mockersf** [comment, marked as answer]
assets are ref counted by their handles, and unloaded when no strong handle link to an asset anymore 👍

**Reply by jeffw387** [reply]
I believe that answers my question, thank you!
